plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -2072,6 +2121,17 @@ protected StartAnswer execute(StartCommand cmd) {
     }
 
     /**
+     * Update boot args with the new nic mac addresses.
+     */
+    protected String replaceNicsMacSequenceInBootArgs(String oldMacSequence, String newMacSequence, VirtualMachineTO vmSpec) {
+        String bootArgs = vmSpec.getBootArgs();
+        if (!StringUtils.isEmpty(bootArgs) && !StringUtils.isEmpty(oldMacSequence) && !StringUtils.isEmpty(newMacSequence)) {
+            return bootArgs.replace(oldMacSequence, newMacSequence);
+        }
+        return "";
```

Comment:
Is this case possible? `bootArgs` not empty and `oldMacSequence` or `newMacSequence` empty

If so, would not it be better to return at line 2131 the variable `bootArgs`?